### PR TITLE
fix(typos): fixed comments, variables anf functions name

### PR DESCRIPTION
**Pull request by rgallor**

### CODE REVIEW

File: examples/individual_datastream/README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ If you have not already done so, make sure that you satisfy the
 Now that our device is connected to Astarte and is sending periodically data we can check if
 the Astarte cluster is storing correctly this data.
 
-We can check if the publish from the device has been correctly received on the server using
+We can check if the publishing from the device has been correctly received on the server using
```
Comment (joshuachp) [inline examples/individual_datastream/README.md:28]: This was correct, meaning the call to publish

File: examples/individual_properties/README.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ If you have not already done so, make sure that you satisfy the
 Upon connecting to the local Astarte instance, the instance SDK will start updating periodically
 the value of one of is properties.
 
-We can check if the publish from the device has been correctly received on the server using
+We can check if the publishing from the device has been correctly received on the server using
```
Comment (joshuachp) [inline examples/individual_properties/README.md:27]: Same as above

File: examples/individual_datastream/README.md (hunk 1 above)
Comment (joshuachp) [reply examples/individual_datastream/README.md:28]: Or MQTT publish

File: src/types.rs [hunk 3]
```diff
@@ -18,7 +18,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 //! Provides Astarte specific types to be used by the
-//! [AstarteDeviceSdk][crate::AstarteDeviceSdk] to transmit/receivedata to/from the Astarte cluster.
+//! [AstarteDeviceSdk][crate::AstarteDeviceSdk] to transmit/receive data to/from the Astarte cluster.
```
Comment (joshuachp) [inline src/types.rs:21]: I would remove data altogether

File: src/types.rs [hunk 4]
```diff
@@ -33,7 +33,7 @@ use crate::{interface::MappingType, Timestamp};
 #[derive(Debug, Clone, thiserror::Error)]
 pub enum TypeError {
     /// Invalid floating point value
-    #[error("forbidden floating point number, Nan, Infinite or subnormals are invalid")]
+    #[error("forbidden floating point number, Nan, Infinite or subnormal are invalid")]
```
Comment (joshuachp) [inline src/types.rs:36]: I would add numbers here then

**Review by joshuachp — CHANGES_REQUESTED** [review]
Other for the comments LGTM, good catches

File: src/types.rs (hunk 4 above)
Comment (rgallor) [reply src/types.rs:36]: Do you mean using `subnormal numbers`?

File: src/types.rs (hunk 3 above)
Comment (rgallor) [reply src/types.rs:21]: Do you mean maintaining only `receive` without `data`?

File: src/types.rs (hunk 4 above)
Comment (joshuachp) [reply src/types.rs:36]: Yess

**Review by joshuachp — APPROVED** [review]

**Review by harlem88 — APPROVED** [review]

**Pull request merged**